### DISCUSSION TITLE
Connection errored out.

### DISCUSSION THREAD

**Discussion by octavioismael-fer** [lllyasviel/Fooocus#3036] [Q&A] [answered]
Hello, I've been having this same error for a few weeks now when I want to generate images.

I've never had it before.

I'm using fooocus from a google colab notebook.

Here I leave an image so you can see the error and I will also share the output of the console.

![Error_fooocus](https://github.com/lllyasviel/Fooocus/assets/91096397/e9a4f14d-cf2c-4cb2-b21d-e0c7220af8a1)

Here's the console's output:

[Parameters] Adaptive CFG = 7
[Parameters] CLIP Skip = 2
[Parameters] Sharpness = 2
[Parameters] ControlNet Softness = 0.25
[Parameters] ADM Scale = 1.5 : 0.8 : 0.3
[Parameters] CFG = 7.0
[Parameters] Seed = 1905599735576349296
[Parameters] Sampler = dpmpp_2m_sde_gpu - karras
[Parameters] Steps = 30 - 15
[Fooocus] Initializing ...
[Fooocus] Loading models ...
Refiner unloaded.
model_type EPS
UNet ADM Dimension 2816
Using pytorch attention in VAE
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
Using pytorch attention in VAE
^C

If anyone knows how to fix this error or why and can comment on it, it would be very helpful.
Thank you

**Comment by ILikeToasters** [comment]
Does it happen after a period of time? If using the free tier of Colab you only get a limited amount of usage per day. You can check her for more info since that could be your issue but not sure. 
<img width="401" alt="Screenshot 2024-05-28 101936" src="https://github.com/lllyasviel/Fooocus/assets/4357657/006f310e-2637-4965-a6da-5d2adc6f8468">

**Comment by octavioismael-fer** [comment]
This time it's happening at the beginning, that is, I launch the script and when I get to the gradio interface (I select the model/lora, input prompt, etc), it doesn't load anything that throws that error at me.

That's right, I'm using an account with colab's free plan.
However, I didn't have that error before.

I checked it to make sure I had time available and it seems like I do.

I ask you a question, are you using an account with a "pro" plan or some plan that is not free?

Do you have this problem too?

![image](https://github.com/lllyasviel/Fooocus/assets/91096397/ae0100ea-788a-471b-80f3-c111d3e83a83)

**Comment by Ckrotky** [comment]
I have the same problem. It started after the last update.
I am using Colab Pro (T4)
When I run it for the first time, everything works. For test - I generate with simple prompt and standard settings - pictures are generated. But when I try to change the model or load Metadata from a file - immediately I get the same error. In the console the same thing. When reloading the page - I get the standard Colab message - “No interface is running right now”

**Reply by ILikeToasters** [reply]
The only difference between your issue and the OP is that he is saying it has been happening for a couple weeks. So if the case would not be related to the update. I will see if I can get the issue to happen for me. I don't use Colab often.

**Reply by Ckrotky** [reply]
oh, yeah. you're right, I wasn't paying attention....
But it's strange that this topic appeared at the very moment when I started having problems too.🤔

**Reply by ILikeToasters** [reply]
I did have it happen to me. just now. But I am not very familiar with Colab so not sure. I do know sometimes the issues have to do with Gradio or something else besides Fooocus. Hopefully someone more familiar will have an idea.

**Reply by Ckrotky** [reply]
Previously I assumed that it was due to insufficient disk space (yesterday I gave up the Colab Pro plan and used only Pay As You Go. After starting Fooocus I had about 12GB free space), because Fooocus can additionally load the necessary data and models directly in the process.
I have just increased the space - to 200Gb - the error repeated. ((


I will try to run the previous version of Fooocus (2.3.1 - it worked for me stably), If the problem repeats - definitely the problem is in Colab, not in Fooocus.

**Reply by octavioismael-fer** [reply]
I'm running Fooocus on Kaggle and I'm not having any issues.
Therefore, I agree that the problem must be in colab.

**Comment by mrboshek** [comment]
Okay, this is odd. It seems like I have to generate one image with juggernaut8 before loading a second model.  For example, if I try to generate my first image with Proteus or a Pony model, then I will get Errored Out.  This started happening yesterday with free Colab.

Update: The above doesn't consistently hold true. Juggernaut 8 always seems to work but not a second model.

**Reply by octavioismael-fer** [reply, marked as answer]
Now that you mention it, I tried working with the base medelo that comes with fooocus and I didn't get any errors, but when I change the model it automatically shows.

Even changing the preset, it has a bug.

Anyway I think it's clear that the problem is with Colab and not with Fooocus as such, since I ran Fooocus in a Kaggle notebook and it worked perfectly.

**Comment by mrboshek** [comment]
I don't know, I think this should start a topic in issues.  I tried a notebook that is using a Fooocus file from Camenduru that is still working with different models.  Too bad it appears to be using Fooocus v 1.0.40.
It's strange, sometimes the new version of Fooocus works properly.  It seems like this latest version isn't playing nicely with free Colab.  Is there a way to use the previous version. I've been using Fooocus very regularly since November.

**Reply by octavioismael-fer** [reply]
Could you share the version you use? that is, that version of Camenduru.

In fact I've been using Fooocus for a while now and I hadn't had these problems before.

**Reply by mrboshek** [reply]
I tried a notebook from this page: https://github.com/camenduru/Fooocus-colab.  I tested a few models using the KandooAI notebook from this page. It worked, but it's version 1.0.43, so it's very old and barebones without the newer fooocus features. Hope to find a more up to date option.

**Comment by Ckrotky** [comment]
Guys, I think I've figured out what the problem is, but I don't know how to solve it. 
I'm using a paid Colab Pro subscription (I think for free Colab accounts the problem will be even more relevant. ). And the problem is really about resources. Specifically RAM.
I use the T4 runtime environment all the time. This gives me 12.7 RAM, 15GB GPU and 200 GB of space 
I've done a measurement. Without changing the model - resources consume about 10GB
![image](https://github.com/lllyasviel/Fooocus/assets/1045831/760fa3af-13b6-4b46-84b6-1993eec8eb9a)

At the moment when I'm in Fooocus trying to switch to a new model and generate something - RAM resources are almost completely used up.
![image](https://github.com/lllyasviel/Fooocus/assets/1045831/a0cebc67-75d0-40c7-9df2-54a8df247c7c)

This may not be related, but I also noticed one more thing. At the same moment in Colab I get a message 

> Use Control-C to stop this server and shut down all kernels (twice to skip confirmation). 

I'm sure that's why at the moment of the error in the console it always says “^C” at the end. To be honest, I don't know what exactly to associate this message with. I assume Python itself is generating it, but I could be mistaken.

Next I decided to change the runtime environment to L4 GPU (62.8 GB RAM / 22.5 GB GPU). After changing the model and trying to generate anything - no errors occurred. Fooocus continued to work. At peak time - RAM consumption was 16GB which is 25% more than what is allocated at T4.
![image](https://github.com/lllyasviel/Fooocus/assets/1045831/aed9677a-3f45-4354-ab30-d010160fc596)

Previously T4 was quite enough for me. But now using L4 GPU is a bit expensive for me (almost T4 x 3), especially since I don't need most of the allocated resources.

Can anyone help what exactly the problem with increasing RAM is related to
And how to solve it?

**Reply by octavioismael-fer** [reply]
Great analysis on resource consumption seems interesting, however from my side I can't help much since I don't have much idea about the matter.

Hopefully someone with more experience on the subject can read this and give us a hand.

**Reply by AnalyticsXXI** [reply]
> Can anyone help what exactly the problem with increasing RAM is related to And how to solve it?

Are you using  "--always-high-vram" flag?

**Comment by ItsMe-Haris** [comment]
I was also facing the same error when using input image. I don't think this error is related to swap size as I used the same input image but in jpg compression and it worked. It worked on the same instance without restarting or anything. I think this is somewhat related to image size as Gradio uses webhook connection.

I haven't figured out size limit yet but using some compression (no resolution decrease) will solve this issue if you're facing one.

**Comment by mrboshek** [comment]
Tested the free Colab version of DeFooocus with a model and Lora I added.  It works. Not sure why the official version of fooocus won't allow model changes for me.  Defooocus seems as up to date as the official version.

**Reply by ILikeToasters** [reply]
I don't think Defooocus has been updated in a few months from looking.

**Reply by mrboshek** [reply]
That's what I thought, but then I noticed their debugger mode had a nsfw blocker similar to the Fooocus update. This was my first time trying Defooocus, so I assumed they updated along with Fooocus. However, I don't see the clip skip option in Defooocus.
I'm scrambling to find a reasonably up to date Colab alternative to Illyasviel's Fooocus that will allow me to change models. Lately, Fooocus has been my go to over my sagemaker a1111 notebook.  I could go back to a1111 as my primary option, but I'd rather stick with fooocus.

**Comment by Ckrotky** [comment]
Guys, I think I found the solution.
If you are using Colab Pro, just select High-RAM.
![image](https://github.com/lllyasviel/Fooocus/assets/1045831/400fe227-e55c-43e5-a39b-b29ce201ba4f)

Unfortunately, I haven't found a solution for the free Colab. (

**Comment by mrboshek** [comment]
Free version is working again for me this morning.  Sample of my notebook below.  I run the first cell, stop when done , then run second cell, click second cell's gradio link. Not sure if this is the best way, but it's currently working.

#cell 1
!pip install pygit2==1.12.2
%cd /content
!git clone https://github.com/lllyasviel/Fooocus.git
%cd /content/Fooocus
!python entry_with_update.py --share --always-high-vram

#cell 2
!curl -Lo /content/Fooocus/models/checkpoints/ponyvision6xl.safetensors https://civitai.com/api/download/models/290640
!curl -Lo /content/Fooocus/models/checkpoints/proteusrundiffusion.safetensors https://huggingface.co/dataautogpt3/Proteus-RunDiffusion/resolve/main/Proteus-RunDiffusion.safetensors?download=true

!wget -c https://civitai.com/api/download/models/135867 -O ./models/loras/detailtweaker.safetensors

%cd /content/Fooocus
!python entry_with_update.py --share --always-high-vram

**Reply by octavioismael-fer** [reply]
Thank you for sharing your experience and your code, I'll try it later and give me my opinion on the subject!

**Reply by mrboshek** [reply]
I disconnected the runtime and restarted to test if it works again. It does!  Maybe it was a temporary Colab issue?  The RAM doesn't have that high spike anymore, as shown above by Ckrotky.
